src/server/dfly_main.cc

```diff
@@ -459,7 +274,11 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
   service.Init(acceptor, listeners, opts);
 
   VersionMonitor version_monitor;
-  version_monitor.Run(pool);
+
+  // check if it's a production release tag.
+  if (GetFlag(FLAGS_version_check) && kGitTag[0] == 'v' && strchr(kGitTag, '-') == nullptr) {
+    version_monitor.Run(pool);
```

Comment:
Are there any changes in the Run function expect from the line above pulled out and moving it to a different file?
Its impossible with github to see if there are any changes in the implementation

Reply:
not, it's just moving the same code.